android/src/main/java/io/ably/flutter/plugin/AblyEventStreamHandler.java

```diff
@@ -0,0 +1,107 @@
+package io.ably.flutter.plugin;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import io.ably.lib.realtime.ChannelStateListener;
+import io.ably.lib.realtime.ConnectionStateListener;
+import io.flutter.plugin.common.EventChannel;
+
+
+public class AblyEventStreamHandler implements EventChannel.StreamHandler {
+    private final AblyMethodCallHandler methodCallHandler;
+
+    AblyEventStreamHandler(AblyMethodCallHandler methodCallHandler){
+        this.methodCallHandler = methodCallHandler;
+    }
+
+    /**
+     * Refer to the comments on AblyMethodCallHandler.MethodResultWrapper
+     * on why this customized EventSink is required
+     * */
+    private static class MainThreadEventSink implements EventChannel.EventSink {
+        private EventChannel.EventSink eventSink;
+        private Handler handler;
+
+        MainThreadEventSink(EventChannel.EventSink eventSink) {
+            this.eventSink = eventSink;
+            handler = new Handler(Looper.getMainLooper());
+        }
+
+        @Override
+        public void success(final Object o) {
+            handler.post(() -> eventSink.success(o));   //lambda for new Runnable
+        }
+
+        @Override
+        public void error(final String s, final String s1, final Object o) {
+            handler.post(() -> eventSink.error(s, s1, o));
+        }
+
+        @Override
+        public void endOfStream() {
+            //TODO work on this if required, or remove this TODO once all features are covered
```

Comment:
As with other PR, can we ensure TODO tasks carry an issue number if in PR?

Reply:
Added https://github.com/ably/ably-flutter/issues/9